src/Compilers/CSharp/Portable/PublicAPI.Unshipped.txt

```diff
@@ -153,6 +153,13 @@ Microsoft.CodeAnalysis.CSharp.Syntax.TupleTypeSyntax.WithCloseParenToken(Microso
 Microsoft.CodeAnalysis.CSharp.Syntax.TupleTypeSyntax.WithElements(Microsoft.CodeAnalysis.SeparatedSyntaxList<Microsoft.CodeAnalysis.CSharp.Syntax.TupleElementSyntax> elements) -> Microsoft.CodeAnalysis.CSharp.Syntax.TupleTypeSyntax
 Microsoft.CodeAnalysis.CSharp.Syntax.TupleTypeSyntax.WithOpenParenToken(Microsoft.CodeAnalysis.SyntaxToken openParenToken) -> Microsoft.CodeAnalysis.CSharp.Syntax.TupleTypeSyntax
 Microsoft.CodeAnalysis.CSharp.Syntax.VariableDesignationSyntax
+Microsoft.CodeAnalysis.CSharp.Syntax.VersionDirectiveTriviaSyntax
+Microsoft.CodeAnalysis.CSharp.Syntax.VersionDirectiveTriviaSyntax.Update(Microsoft.CodeAnalysis.SyntaxToken hashToken, Microsoft.CodeAnalysis.SyntaxToken versionKeyword, Microsoft.CodeAnalysis.SyntaxToken endOfDirectiveToken, bool isActive) -> Microsoft.CodeAnalysis.CSharp.Syntax.VersionDirectiveTriviaSyntax
```

Comment:
This should not require any API changes, I think. A `#version 7.1` directive is already an error in the lexer, where all of the version information (language and assembly) is available. I think the implementation of this pseudo-feature should just be to change the error that is produced when that is recognized, and leave the syntax tree the same as it would have been previously, with only the diagnostic changed.